apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -552,20 +584,43 @@ private void AssertOpen(string message)
                 throw new InvalidOperationException("The connection must be open.  Failed to " + message);
         }
 
-        private Task<SpannerTransaction> BeginTransactionImpl(CancellationToken cancellationToken,
-            TransactionOptions transactionOptions,
-            TransactionMode transactionMode)
+        internal Task<SingleUseTransaction> BeginSingleUseTransactionAsync(
+            TimestampBound targetReadTimeStamp,
+            CancellationToken cancellationToken)
+        {
+            var options = ConvertToOptions(targetReadTimeStamp);
+            return ExecuteHelper.WithErrorTranslationAndProfiling(async () =>
+            {
+                using (var sessionHolder = await SessionHolder.Allocate(this,
+                        options, cancellationToken)
+                    .ConfigureAwait(false))
+                {
+                    await sessionHolder.Session.RemoveFromTransactionPoolAsync();
```

Comment:
I did another pass and found some more missing configures, so we should do this at some point.